app/models/list.model.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * (c) 2016 Tieto Finland Oy
+ * Licensed under the MIT license.
+ */
+'use strict';
+
+/**
+* @ngdoc service
+* @name dashboard.ListData
+* @description
+* # ListData
+* Model in the dashboard.
+*/
+angular.module('dashboard')
+    .factory('ListData', ['AttachmentData', function (AttachmentData) {
+
+        function ListData(title, objects) {
+            this.title = title;
+            this.objects = objects;
+        }
+
+        ListData.createAttachmentList = function (title, attachment) {
+            if (typeof title === 'string') {
+                var array = [];
+                for (var index = 0; (attachment instanceof Array) && (index < attachment.length); index++) {
```

Comment:
Maybe move attachment type checking together with title type checking, no need to check on each loop iteration?